src/components/common/ProtectedRoute/index.tsx

```diff
@@ -0,0 +1,18 @@
+import { User } from 'api/user/entity';
+import { Navigate, Outlet } from 'react-router-dom';
+
+interface Props {
+  auth: User | null;
+  redirectPath?: string;
+}
+
+export default function ProtectedRoute({
+  auth,
+  redirectPath = '/',
+}: Props) {
+  if (auth) {
+    return <Navigate to={redirectPath} replace />;
+  }
+
+  return <Outlet />;
+}
```

Comment:
(아래 리뷰 사항이 반영됐다고 가정)
굳이 auth 를 주입받을 필요가 없어 보여요.

```suggestion
export default function ProtectedRoute({ redirectPath = '/' }: Props) {
  const auth = useAuthValue();

  if (auth) {
    return <Navigate to={redirectPath} replace />;
  }

  return <Outlet />;
}
```

Reply:
`App.tsx`에서 auth의 상태를 관리하는 부분이 불필요했네요 지금보니 어울리지도 않구요. 수정했습니다 😄